### DISCUSSION TITLE
remix cache headers get ignored

### DISCUSSION THREAD

**Discussion by marcelmusiol** [vercel/community#1559] [Help] [answered]
### Summary

not sure if this is a problem with vercel or remix

### Steps to reproduce:
create new remix app with `npx create-remix@latest`
deploy target vercel

in `entry.server` include
```js
responseHeaders.set("Cache-Control", "s-maxage=180")
```

### expected behaviour:
the vercel edge cache should cache the document for 180seconds

### actual behaviour:
the document is never cached
`X-Vercel-Cache: MISS`

```
dependencies:
+ @remix-run/node 1.13.0
+ @remix-run/react 1.13.0
+ @remix-run/vercel 1.13.0
+ @vercel/node 2.9.6
+ react 18.2.0
+ react-dom 18.2.0

devDependencies:
+ @remix-run/dev 1.13.0
+ @remix-run/eslint-config 1.13.0
+ @remix-run/serve 1.13.0
+ @types/react 18.0.28
+ @types/react-dom 18.0.11
+ eslint 8.34.0
+ typescript 4.9.5
```

**Comment by chaunceyau** [comment]
+1, this notably just started happening. had a deploy from 89 days ago that worked fine, deployed a fresh version of the same exact commit and now it's broken.

I use the headers function export to set headers and they are coming through in the request hitting my browser. Also getting the `x-vercel-cache: MISS`

notably, these two urls were deployed from the exact same git commit..
- cache not working (top deployment - 2h ago): https://chief-austinc-remix-edge-cache-demo-eooeppx0o-achauncey.vercel.app
- cache working (2nd deployment -  89d ago): https://chief-austinc-remix-edge-cache-demo-3qu9y5571-achauncey.vercel.app
- source code: https://github.com/chaunceyau/chief-austinc-remix-edge-cache-demo
- source commit: [3a18860](https://github.com/chaunceyau/chief-austinc-remix-edge-cache-demo/commit/3a18860b8df2bbfb49548acee1f885bb05eb60d3)

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/32502305/219984407-eedad626-eb8d-4862-95d7-5c3dd23b7784.png">

**Comment by TooTallNate** [comment]
Thank you for the report. Confirmed that this is an issue, we are looking into it.

**Reply by maxcountryman** [reply]
@TooTallNate should this be listed as a partial outage on the Vercel status page?

I’m running into this as well and it’s unfortunate that there isn’t any public-facing mention of this beyond your comment here.

**Reply by TooTallNate** [reply]
It's not so much an outage, but it was a change to the Remix builder that uncovered this edge case at the infra level.

If you need an immediate workaround, set the environment variable `VERCEL_CLI_VERSION` with a value of `vercel@28.15.4` on your deployment. But the issue has been identified and a fix should roll out soon, hopefully by tomorrow.

**Reply by maxcountryman** [reply]
@TooTallNate I'm seeing the same issue with a project that isn't Remix. It seems like the CDN is mangling Cache-Control headers in unexpected ways. That certainly sounds like an outage from where I'm standing.

**Reply by leerob** [reply]
@maxcountryman we have not seen any other reports outside of Remix. Do you have a reproduction or more information?

**Reply by maxcountryman** [reply]
@leerob I have an active support ticket. I haven't heard back since yesterday. I'm seeing the behavior described here as well as mangling of the headers. Unclear if these are the same issue, however it is clear that the Vercel CDN is not setting these headers correctly.

**Reply by leerob** [reply]
Thanks for letting us know, I'll chat with the support team. If you're on a free plan, the responses are a bit slower than paid plans, just an FYI. If you can provide steps towards reproduction where you're seeing this issue on the support ticket, our team can investigate further to see if it's related to this issue.

**Reply by maxcountryman** [reply]
Thank you, I appreciate that. I shared one deployment showing the issue on the ticket as well as the relevant config. Happy to provide any additional info to help diagnose.

**Comment by edgarasben** [comment]
I am having the same issue. HTTP cache headers are ignored on a new fresh install of Remix.

Repo: https://github.com/edgarasben/test-remix
Demo: https://test-remix-xi.vercel.app/

**Reply by maxcountryman** [reply]
I'm seeing this with Astro as well. So far cannot get a response from support. I would really like to see this on the status page for Vercel if this is indeed an issue with the surrogate caching proxy, i.e. the Vercel CDN itself. /cc @leerob

**Comment by TooTallNate** [comment, marked as answer]
The fix has been pushed to production. I see the caching working as expected on i.e. https://chief-austinc-remix-edge-cache-demo-eooeppx0o-achauncey.vercel.app/

Please let us know if you are having issues.

**Reply by edgarasben** [reply]
Seems to be working now! 👍

**Reply by marcelmusiol** [reply]
I can confirm too, it's working, even without redeploying. Thank you @TooTallNate

**Reply by marcelmusiol** [reply]
@TooTallNate just checked again and it seems to be broken again

**Reply by TooTallNate** [reply]
Sorry, unfortunately it appears that we needed to disable it temporarily. I'll let you know when it's turned back on.

**Reply by TooTallNate** [reply]
From what I'm told, this was fixed for good at this point. Sorry for the hiccup.

**Reply by marcelmusiol** [reply]
Yes, looking good. Thanks Nathan!

**Comment by Coretteket** [comment]
Wanted to jump in and note that the same is happening for my SvelteKit app. This issue is causing quite a significant regression on my deployments, and it is not solved by the earlier mentioned `VERCEL_CLI_VERSION` environment variable.

**Reply by maxcountryman** [reply]
Would really like Vercel to comment on this and again, I have to stress, I would categorize this as a partial outage of the CDN.

**Reply by Coretteket** [reply]
Cache headers appear to be working again – at least on my latest deployment. @TooTallNate, has this indeed been fixed, or could it break again soon?

**Reply by TooTallNate** [reply]
From what I'm told, this was fixed for good at this point.

**Reply by Coretteket** [reply]
Thanks, appreciate the update!
